### PR TITLE
Make tbptt imports Python 3.10 compatible

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

`collections.Sequence` got moved to `collections.abc.Sequence` in Python 3.10. 
This fixes the code paths for TBPTT when running in Python 3.10. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @akihironitta

### CODE REVIEW

File: src/pytorch_lightning/core/module.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 """The LightningModule - an nn.Module with many additional features."""
 
-import collections
+import collections.abc
```
Comment (kaushikb11) [inline src/pytorch_lightning/core/module.py:16]: importing just `collections` should be enough, no?

**Review by justusschock — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: src/pytorch_lightning/core/module.py (hunk 1 above)
Comment (justusschock) [reply src/pytorch_lightning/core/module.py:16]: no, as given [here](https://docs.python.org/3.9/library/collections.html#module-collections):
> Deprecated since version 3.3, will be removed in version 3.10: Moved [Collections Abstract Base Classes](https://docs.python.org/3.9/library/collections.abc.html#collections-abstract-base-classes) to the [collections.abc](https://docs.python.org/3.9/library/collections.abc.html#module-collections.abc) module. For backwards compatibility, they continue to be visible in this module through Python 3.9.

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/core/module.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/core/module.py:16]: I did this to avoid the huge squiggly lines in the IDE :)
